app/models/katello/erratum.rb

```diff
@@ -92,28 +92,20 @@ def self.installable_for_hosts(hosts = nil)
     end
 
     def self.ids_installable_for_hosts(hosts = nil)
-      hosts = ::Host.where(:id => hosts) if hosts && hosts.is_a?(Array)
-
       # Main goal of this query
       # 1) Get me the applicable errata for these set of hosts
       # 2) Now further prune this list. Only include errata from repos that have been "enabled" on those hosts.
       #    In other words, prune the list to only include the errate in the "bound" repositories signified by
       #    the inner join between ContentFacetRepository and RepositoryErratum
-      query = self.joins(:content_facet_errata).
-        joins("INNER JOIN #{Katello::ContentFacetRepository.table_name} on \
-        #{Katello::ContentFacetRepository.table_name}.content_facet_id = #{Katello::ContentFacetErratum.table_name}.content_facet_id").
-        joins("INNER JOIN #{Katello::RepositoryErratum.table_name} AS host_repo_errata ON \
-          host_repo_errata.erratum_id = #{Katello::Erratum.table_name}.id AND \
-          #{Katello::ContentFacetRepository.table_name}.repository_id = host_repo_errata.repository_id")
+      facet_repos = Katello::ContentFacetRepository.joins(:content_facet => :host).select(:repository_id).distinct
+      facet_errata = Katello::ContentFacetErratum.joins(:content_facet => :host).select(:erratum_id).distinct
```

Comment:
The `.distinct` causes a sort to happen, and if the size of the sort working set is larger than postgres's `work_mem`, it will write to temporary files. This can be a big deal (and even cause outages from running out of disk space) if the sort is large enough, especially if the query is slow and the user refreshes the page a few times.

I don't have any issue with the distinct itself, but you may want to try it out with a very large DB and enable `log_temp_files` in postgresql.conf.